Lidgren.Network/NetUPnP.cs

```diff
@@ -157,19 +162,22 @@ internal void ExtractServiceUrl(string resp)
 
 			XmlNamespaceManager nsMgr = new XmlNamespaceManager(desc.NameTable);
 			nsMgr.AddNamespace("tns", "urn:schemas-upnp-org:device-1-0");
-			XmlNode typen = desc.SelectSingleNode("//tns:device/tns:deviceType/text()", nsMgr);
-			if (!typen.Value.Contains("InternetGatewayDevice"))
+			XmlNode? typen = desc.SelectSingleNode("//tns:device/tns:deviceType/text()", nsMgr);
+			if (typen?.Value?.Contains("InternetGatewayDevice") != true)
 				return;
 
 			var serviceName = "WANIPConnection";
-			XmlNode node = desc.SelectSingleNode("//tns:service[tns:serviceType=\"urn:schemas-upnp-org:service:" + serviceName + ":1\"]/tns:controlURL/text()", nsMgr);
+			XmlNode? node = desc.SelectSingleNode("//tns:service[tns:serviceType=\"urn:schemas-upnp-org:service:" + serviceName + ":1\"]/tns:controlURL/text()", nsMgr);
 			if (node == null)
 			{
 				//try another service name
 				serviceName = "WANPPPConnection";
 				node = desc.SelectSingleNode("//tns:service[tns:serviceType=\"urn:schemas-upnp-org:service:" + serviceName + ":1\"]/tns:controlURL/text()", nsMgr);
-				if (node == null)
-					return;
+			}
+
+			if (node?.Value == null)
+			{

```

Comment:
Error reporting in this code is pretty bad anyways, but this change makes it worse. Now if there's more issues receiving the UPnP response, it'll silently fail instead of logging something.